### PR TITLE
Bach: SeriesString.str.replace

### PR DESCRIPTION
**Pull request by KathiaBarahona**

### CODE REVIEW

File: bach/tests/functional/bach/test_series_string.py [hunk 1]
```diff
@@ -110,3 +110,24 @@ def test_get_dummies(engine) -> None:
...
         ],
     )
+
+
+def test_string_replace(engine) -> None:
+    bt = get_df_with_test_data(engine)
+    municipality = bt['municipality'].sort_index()
+    assert isinstance(municipality, SeriesString)
+
+    municipalities = municipality.to_numpy()
```
Comment (thijs-obj) [inline bach/tests/functional/bach/test_series_string.py:120]:
```suggestion
    municipalities = ['Leeuwarden', 'Súdwest-Fryslân', 'Súdwest-Fryslân']
```
Doing `to_numpy()` involves an extra query, which takes ~1.5 seconds for BigQuery. Considering in how many places we have hardcoded the expected values already, I don't think it's bad to do here, and it does safe run-time (and personally, I find it more readable actually)

File: bach/tests/functional/bach/test_series_string.py [hunk 2]
```diff
@@ -110,3 +110,24 @@ def test_get_dummies(engine) -> None:
...
+    replace_pat = ['-', 'west', 'ee', 'ú', 'â']
```
Comment (thijs-obj) [inline bach/tests/functional/bach/test_series_string.py:121]:
```suggestion
    replace_pat = ['-', 'west', 'ee', 'ú', 'â', 'e']
```
Having at least one case where we replace more than one occurrence in the string seems like a good idea to me

**Review by thijs-obj — COMMENTED** [review]
Code looks good to me. Some comments on the tests

File: bach/tests/functional/bach/test_series_string.py (hunk 1 above)
Comment (KathiaBarahona) [reply bach/tests/functional/bach/test_series_string.py]: Done here https://github.com/objectiv/objectiv-analytics/pull/953/commits/110bcaa49477d0797201c744fa268b548198d474

**Pull request merged**